VirtoCommerce.Storefront/Domain/Security/SecurityWorkContextBuilderExtensions.cs

```diff
@@ -26,7 +27,19 @@ public static async Task WithCurrentUserAsync(this IWorkContextBuilder builder)
             }).ToList();
 
             var user = await signInManager.UserManager.GetUserAsync(builder.HttpContext.User);
-           
+
+            if (user != null && !new CanUserLoginToStoreSpecification(user).IsSatisfiedBy(builder.WorkContext.CurrentStore))
```

Comment:
I've removed these lines by this commit https://github.com/VirtoCommerce/vc-storefront-core/commit/c436160807c6a07d4ce2e01ae95b1ad28fdb1654#diff-4572e31bd7dfce2f9afdc19c23dd0881L30  but unfortunately, I didn't write a unit-tests for this bug, but  I remember that this was due to problems with automatic user logout when using several stores. I would like to recommend removing these lines until we won't ensure that all works as expected.

Reply:
@tatarincev We may replace it with Forbidden response but we need to prevent customer login into another store anyway.

Reply:
Or I think it's better to remove sign out and leave user as anonymous